src/web/resources/report.js

```diff
@@ -35,91 +35,135 @@ function Element(tagname, props, children) {
+    attachLeaderToChildren: function (leaderTag, leaderName, childNodes) {
+        const parentLabel = this.buildCheckboxLabel(leaderTag, leaderName, true)
+        parentLabel.addEventListener("click", () => {
+            const parentState = parentLabel.querySelector("input").checked
+            childNodes.forEach((child) => {
+                const children = this.getRowsForClass(child.classList[0])
+                child.querySelector("input").checked = parentState
+                this.updateChildren(children, parentState)
+                this.reDrawGraph()
             })
-        }
+        })
+        return parentLabel
     },
-    createChild: function (childName) {
-        const count = document.querySelectorAll(`tr.${childName}`)
-        const childBox = this.buildCheckboxLabel(childName, `${renames[childName]} (${count.length})`, true)
-
-        childBox.addEventListener("click", (e) => {
+    createChild: function (childName, count) {
+        const childNode = this.buildCheckboxLabel(childName, `${renames[childName]} (${count})`, true)
+        childNode.addEventListener("click", (e) => {
             const thisChild = e.target.querySelector("input")
             if (thisChild == null) { return }
-            this.updateDomNodesWithID(childName, !thisChild.checked)
+            const childr = this.getRowsForClass(childName)
+            this.updateChildren(childr, !thisChild.checked)
+            this.reDrawGraph()
         })
-        return childBox
+        return childNode
     },
-    // helper function to loop over the table and toggle state of nodes 
-    // with `#stringID`
-    updateDomNodesWithID: function (stringID, state) {
-        const siblingsList = document.querySelectorAll(`#results tbody tr`)
-        siblingsList.forEach((child, n, p) => {
-            if (child.classList.contains(stringID)) {
-                if (state) {
-                    child.classList.remove("hidden-row")
-                } else {
-                    child.classList.add("hidden-row")
+    startingTestData: async function () {
+        if (resultsTestData == null) {
+            let response = await fetch("results.json", {
+                headers: { "content-type": "text/plain" },
+                method: "GET",
+                mode: "cors",
+            })
+            resultsTestData = (await response.json()).tests
+            return resultsTestData
+        } else {
+            return resultsTestData
+        }
+    },
+    reDrawGraph: async function () {
+        const resultPlot = document.querySelector(ResultPlot.selector)
+        try {
+            const tests = await this.startingTestData()
+            const svg = resultPlot.querySelector("svg")
+            var viewable = []
+            const selected = document.querySelectorAll("#filters > details label > input:checked")
+            selected.forEach((child) => {
+                viewable.push(child.parentNode.classList[0])
```

Comment:
again I know this is bad but not sure how else to dynamically filter based on the classes.

Reply:
maybe I could append "result-"<class-name> to the classes.